src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -153,23 +153,26 @@ ToolsetDefinitionLocations locations
             string overrideTasksPathFromRegistry = null;
             string defaultOverrideToolsVersionFromRegistry = null;
 
+#if FEATURE_WIN32_REGISTRY
             if ((locations & ToolsetDefinitionLocations.Registry) == ToolsetDefinitionLocations.Registry)
             {
-#if FEATURE_WIN32_REGISTRY
-                if (NativeMethodsShared.IsWindows || registryReader != null)
-                // If we haven't been provided a registry reader (i.e. unit tests), create one
-                registryReader = registryReader ?? new ToolsetRegistryReader(environmentProperties, globalProperties);
-
-                // We do not accumulate properties when reading them from the registry, because the order
-                // in which values are returned to us is essentially random: so we disallow one property
-                // in the registry to refer to another also in the registry
-                defaultToolsVersionFromRegistry = registryReader.ReadToolsets(toolsets, globalProperties,
-                    initialProperties, false /* do not accumulate properties */, out overrideTasksPathFromRegistry,
-                    out defaultOverrideToolsVersionFromRegistry);
-            }
+
+                if (NativeMethodsShared.IsWindows || registryReader != null) 
```

Comment:
Is this runtime `IsWindows` check required here, since `FEATURE_WIN32_REGISTRY` should be enabled only for windows/full framework builds anyway? And line 163 (`registryReader = registryReader ?? new ToolsetRegistryReader(...` means that we don't need to check for `registryReader != null` either.

And so, the new set of braces being added here are not required.